crates/sui-core/src/authority/authority_store.rs

```diff
@@ -1294,21 +1206,18 @@ impl AuthorityStore {
             .find_object_lt_or_eq_version(object_id, version)
     }
 
-    /// Returns the last entry we have for this object in the parents_sync index used
-    /// to facilitate client and authority sync. In turn the latest entry provides the
-    /// latest object_reference, and also the latest transaction that has interacted with
-    /// this object.
+    /// Returns the latest object reference we have for this object_id in the objects table.
     ///
-    /// This parent_sync index also contains entries for deleted objects (with a digest of
-    /// ObjectDigest::deleted()), and provides the transaction digest of the certificate
-    /// that deleted the object. Note that a deleted object may re-appear if the deletion
-    /// was the result of the object being wrapped in another object.
+    /// The method may also return the reference to a deleted object with a digest of
+    /// ObjectDigest::deleted() and lamport version of a transaction that deleted the object.
```

Comment:
it can also return ObjectDigest::wrapped(), i think

Reply:
not with current implementation. 
I can extend StoreObject enum to differentiate deleted and wrapped objects if you think we'd need that in the future
to something like
```
pub enum StoreObjectV1 {
    Value(StoreObjectValue),
    Deleted,
    Wrapped,
} 
```
But current call sites(including pruner) do not require that.
Maybe it'd be helpful for debugging